### PR TITLE
Various "fixes" form SR

### PR DESCRIPTION
**Pull request by medariox**
Needs review before merge.

### CODE REVIEW

File: sickbeard/clients/download_station_client.py [hunk 1]
```diff
@@ -31,20 +36,73 @@ def __init__(self, host=None, username=None, password=None):
...
 
+    def _get_auth(self):
+        if self.session.cookies and self.auth:
+            return self.auth
+
+        params = {
+            'api': 'SYNO.API.Auth',
+            'version': 2,
+            'method': 'login',
+            'account': self.username.encode('utf-8') if isinstance(self.username, unicode) else self.username,
```
Comment (labrys) [inline sickbeard/clients/download_station_client.py]: Shouldn't need to be encoded, requests already encodes all params.

File: sickbeard/clients/download_station_client.py [hunk 2]
```diff
@@ -31,20 +36,73 @@ def __init__(self, host=None, username=None, password=None):
...
+            'passwd': self.password.encode('utf-8') if isinstance(self.password, unicode) else self.password,
```
Comment (labrys) [inline sickbeard/clients/download_station_client.py]: Same.

File: sickbeard/clients/download_station_client.py [hunk 3]
```diff
@@ -31,20 +36,73 @@ def __init__(self, host=None, username=None, password=None):
...
+            'session': 'DownloadStation',
+            'format': 'cookie'
+        }
+
+        try:
+            self.response = self.session.get(self.urls['login'], params=params, verify=False)
+            self.response.raise_for_status()
+        except Exception as error:
```
Comment (labrys) [inline sickbeard/clients/download_station_client.py]: this should be:
```python
except RequestException as error:
```
which would also require: `from requests.exceptions import RequestException`

File: sickbeard/clients/download_station_client.py [hunk 4]
```diff
@@ -69,14 +128,89 @@ def _add_torrent_file(self, result):
...
+            'version': 2,
+            'method': 'getinfo',
+            'session': 'DownloadStation'
+        }
 
-        return self.response.json()['success']
+        try:
+            self.response = self.session.get(self.urls['dsminfo'], params=params, verify=False, timeout=120)
+            self.response.raise_for_status()
+        except Exception as error:
```
Comment (labrys) [inline sickbeard/clients/download_station_client.py]: `RequestException`

File: sickbeard/clients/download_station_client.py [hunk 5]
```diff
@@ -69,14 +128,89 @@ def _add_torrent_file(self, result):
...
+                    #  Or forcefully set the location setting in SickRage
+                    params.update({
+                        'method': 'getconfig',
+                        'version': 2
+                    })
+
+                    try:
+                        self.response = self.session.get(self.urls['info'], params=params, verify=False, timeout=120)
+                        self.response.raise_for_status()
+                    except Exception as error:
```
Comment (labrys) [inline sickbeard/clients/download_station_client.py]: `RequestException`

**Comment by medariox** [comment]
Done, thanks @labrys

File: sickbeard/clients/download_station_client.py (hunk 1 above)
Comment (miigotu) [inline sickbeard/clients/download_station_client.py]: sure, requests will encode it with ascii, nice when someone uses non-ascii in a username or password.

**Comment by miigotu** [comment]
If you dont stop changing the committer on my commits when you pirate them there's going to be issues. Take my credit on commits you can leave as is but then put bugs in my commits and leave the committer? Take a class.

**Comment by medariox** [comment]
@miigotu 
You are the committer of unchanged commits, the modified commits are changed as required. I don't know what you are complaining about, but you really gotta stop harassing us in the PRs we open.

**Comment by miigotu** [comment]
Its not harassing. Give credit where it is due. Or even better, write your own code

**Comment by medariox** [comment]
I will make sure we cherry-pick your buggy commits and push a separate fix for them in the future, so you can have all the credit you want for your bugs. Now stop your nonsense please.

**Comment by miigotu** [comment]
If my commits are so buggy, why use them at all? You guys wanted to work on your own project, didnt want to work with me, and dont need me right? Ok thats fine, you left with >1k commits of mine already (about all of your commits combined?).... Really need to take more of my work? But I thought you dont need me and my code is so buggy....

Stand on your own two feet if you want to prove a point, stop piggy backing me.

**Comment by labrys** [comment]
Please keep the PR on topic

File: sickbeard/clients/download_station_client.py (hunk 1 above)
Comment (labrys) [inline sickbeard/clients/download_station_client.py]: @miigotu requests already encodes all params to utf-8
https://github.com/kennethreitz/requests/blob/master/requests/models.py#L95

Comment (labrys) [inline sickbeard/clients/download_station_client.py]: As shown in this example it produces identical output for `.encode` or unicode
```python
>>> requests.get('http://google.com', params={u'ã'.encode('utf-8'): u'ã'}).request.url
'http://www.google.com/?%C3%A3=%C3%A3'
```

**Comment by labrys** [comment]
Approved!

**Pull request merged**